File: Week 03/id_372/best_time_to_buy_and_sell_stock_ii/best_time_to_buy_and_sell_stock_ii.go
package best_time_to_buy_and_sell_stock_ii

func maxProfit(prices []int) int {
	mp := 0
	for i := 0; i < len(prices)-1; i++ {
		if prices[i+1] > prices[i] {
			mp = mp + prices[i+1] - prices[i]
		}
	}

	return mp
}

File: Week 03/id_277/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit = 0;
        for i in range(1,len(prices)):
            if prices[i] > prices[i-1]:
                maxprofit += prices[i]-prices[i-1]
        return maxprofit